src/service.py

```diff
@@ -449,12 +450,34 @@ class DCGMExporter(SnapExporter):
 
     exporter_name: str = "dcgm"
     port: int = 9400
+    metrics_location: Path = Path("/var/snap/dcgm/common/")
+    metric_config: str = "dcgm-exporter-metrics-file"
 
-    def __init__(self, config: ConfigData):
+    def __init__(self, charm_dir: Path, config: ConfigData):
         """Init."""
         self.strategy = DCGMExporterStrategy(str(config["dcgm-snap-channel"]))
+        self.charm_dir = charm_dir
+        self.metrics_file = self.charm_dir / Path("src/gpu_metrics/dcgm_metrics.csv")
+        self.metric_config_value = self.metrics_file.name
         super().__init__(config)
 
+    def install(self) -> bool:
+        """Install the DCGM exporter and configure custom metrics."""
+        if not super().install():
+            return False
+
+        if self.snap_client.get(self.metric_config) != self.metric_config_value:
+            try:
+                shutil.copy(self.metrics_file, self.metrics_location)
+                self.snap_client.set({self.metric_config: self.metric_config_value})
+                self.snap_client.restart(reload=True)
+            except Exception as err:  # pylint: disable=broad-except
+                logger.error("Failed to configure custom DCGM metrics")
+                logger.error("Failed to copy the metrics file: %s", err)
```

Comment:
We don't realy know what failed here so let's be generic
```suggestion
                logger.error("Failed to configure custom DCGM metrics")
```